src/internal_events/datadog_agent.rs

```diff
@@ -0,0 +1,36 @@
+use crate::source_sender::ClosedError;
+use metrics::counter;
+use vector_core::internal_event::InternalEvent;
+
+#[derive(Debug)]
+pub struct DatadogAgentStreamError {
+    pub error: ClosedError,
+    pub count: usize,
+}
```

Comment:
As indicated in my previous review, I think this should be a generic error for all sources.